capsules/src/console.rs

```diff
@@ -204,15 +205,18 @@ impl<'a, U: UART> Client for Console<'a, U> {
         // Either print more from the AppSlice or send a callback to the
         // application.
         self.tx_buffer.replace(buffer);
-        self.in_progress.take().map(|appid| {
+        self.in_progress.get().map(|appid| {
+            self.in_progress.set(None);
```

Comment:
@brghena and I were talking about this. It seems like this is a downgrade from a usage perspective. We've gone from `.map(||)` simply doing the right thing to `.get().map(||)` + don't forget to `set(None)`.

Could `in_progress` be in a MapCell? Or do we need yet a third Cell type?